nixos/modules/services/web-apps/turtl-server.nix

```diff
@@ -0,0 +1,385 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.turtl-server;
+
+  turtlServerConfYAML = pkgs.writeTextFile {
+    name = "turtl-server-config.yaml";
+    text = ''
+      ---
+    '' + lib.optionalString (cfg.listenAddress != null) ''
+      server:
+        host: ${cfg.listenAddress}
+        port: ${toString cfg.listenPort}
+    '' + lib.optionalString (cfg.listenAddress == null) ''
+      server:
+        host:
+        port: ${toString cfg.listenPort}
+    '' + ''
+      db:
+        connstr: "postgres://${cfg.localDatabaseUser}:${cfg.localDatabasePassword}@localhost:5432/${cfg.localDatabaseName}"
+        pool: 24
+
+      loglevel: 'debug'
+
+      app:
+        enable_bookmarker_proxy: false
+        api_url: ${cfg.apiUrl}
+        www_url: ${cfg.wwwUrl}
+        emails:
+          admin: ${cfg.email.admin}
+          info: ${cfg.email.info}
+          invites: ${cfg.email.invites}
+        secure_hash_salt: ${cfg.secureHashSalt}
+        allow_unconfirmed_invites: false
+
+      sync:
+        max_bulk_sync_records: 32
+
+      plugins:
+        plugin_location: ${cfg.plugin.location}
+        analytics:
+          enabled: ${toString cfg.plugin.analytics}
+        email:
+          enabled: ${toString cfg.plugin.email}
+        sync:
+          enabled: ${toString cfg.plugin.sync}
```

Comment:
This will result in `sync: enabled: 1`, is that right? You can use `lib.boolToString` to get the strings `"false"` or `"true"`